client/src/legacy/UnsavedChangesIndicator/UnsavedChangesIndicatorEntwine.js

```diff
@@ -0,0 +1,72 @@
+/* global window */
+import jQuery from 'jquery';
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import { loadComponent } from 'lib/Injector';
+
+jQuery.entwine('ss', ($) => {
+  $('.cms-edit-form .unsaved-changes-indicator__container').entwine({
+    ReactRoot: null,
+    IsDirty: false,
+    Component: null,
+
+    onmatch() {
+      this._super();
+      let root = this.getReactRoot();
+      if (!root) {
+        root = createRoot(this[0]);
+        this.setReactRoot(root);
+      }
+      this.renderComponent();
+    },
+
+    onunmatch() {
+      this._super();
+      const root = this.getReactRoot();
+      if (root) {
+        root.unmount();
+        this.setReactRoot(null);
+      }
+    },
+
+    renderComponent() {
+      let Component = this.getComponent();
+      if (!Component) {
+        Component = loadComponent('UnsavedChangesIndicatorTimer');
+        this.setComponent(Component);
+      }
+      const props = {
+        minutes: JSON.parse(this.attr('data-minutes')),
+        isDirty: this.getIsDirty(),
+      };
+      const root = this.getReactRoot();
+      root.render(<Component {...props} />);
+    },
+
+    updateDirtyState(isDirty) {
+      this.setIsDirty(isDirty);
+      this.renderComponent();
+    },
+  });
+
+  // Monitor the form for dirty state changes
+  // "changed" class is added by jquery.changetracker.js
+  $('.cms-edit-form').entwine({
+    onmatch() {
+      this._super();
+      const container = this.find('.unsaved-changes-indicator__container').first();
+      if (!container.length) {
+        return;
+      }
+      const self = this;
+      const observer = new MutationObserver(() => {
+        const isChanged = self.hasClass('changed');
+        container.updateDirtyState(isChanged);
+      });
+      observer.observe(this[0], {
+        attributes: true,
+        attributeFilter: ['class'],
+      });
```

Comment:
Why use a mutation observer? You can just use `form.changed` onmatch and onunmatch entwine handlers (entwine is already using a mutation observer under the hood for onmatch)